protos/core.proto

```diff
@@ -23,6 +23,8 @@ message ProjectConfig {
   string schema_suffix = 7;
   string table_prefix = 11;
 
+  string version = 17;
```

Comment:
Done, good point

Reply:
IMO calling it `version` makes sense if that's what we are calling the field in the YAML representation, and as we already have a `dataform_core_version` field that is specified in `CompiledGraph`, given the ProjectConfig is a proto representation of that file.

Reply:
I don't really follow the point about `CompiledGraph`. I don't understand why we can't have two different fields, in two different contexts, that mean different things.

`version` is still confusing. Again, if I see `version` in a workflows YAML file, I am going to assume that this somehow refers to the version _of that workflow_, not the version of the code that interprets it.

Reply:
My original thinking was to deprecate specifying the version in https://github.com/dataform-co/dataform/blob/main/protos/core.proto#L296, but because we're not writing any output to ProjectConfig I don't think there's harm in having the Dataform Core version duplicated in some scenarios

Reply:
So again - this is _not_ duplication. One is a user request, the other is what actually happened. The two may not be the same!

Reply:
I dislike the idea that could happen, but it makes sense